### PR TITLE
Error handlers, morgan, joi and joi-validation middlewares were added

### PR DESCRIPTION
**Pull request by CurlyAnastasi**
Changes:

- some middlewares such as morgan and joi-validation were added to order route
- validation of data now is in middlewares files, not in model
- now user data is taken from headers
- we validate users also by password
- error handler was added

### CODE REVIEW

File: demo2/app.js [hunk 1]
```diff
@@ -2,18 +2,16 @@ const express = require('express');
...
+app.use(erorHandler);
 
-
-app.listen('3000', () => {
-    console.log('http://localhost:3000/products');
-    console.log('http://localhost:3000/products/45052448');
-    console.log('http://localhost:3000/order');
-})
\ No newline at end of file
+app.listen('3000');
```
Comment (ghost) [inline demo2/app.js]: Better provide constant for port.

File: demo2/app.js [hunk 2]
```diff
@@ -2,18 +2,16 @@ const express = require('express');
 const app = express();
 const productsRoutes = require('./routes/productsRoute');
 const orderRoutes = require('./routes/orderRoutes');
+const morgan = require('morgan');
+const erorHandler = require('./common/errors/errorsHandler');
```
Comment (ghost) [inline demo2/app.js]: Typo in `errorHandler`

File: demo2/routes/orderRoutes.js [hunk 3]
```diff
@@ -1,9 +1,13 @@
 const express = require('express');
 const orderRoutes = express.Router();
 const orderController = require('../controllers/orderController');
+const validator = require('express-joi-validation').createValidator({})
+const authMidlware = require('../common/midlwares/authMidlware');
```
Comment (ghost) [inline demo2/routes/orderRoutes.js]: Typo in `authMiddleware`

File: demo2/models/orderModel.js [hunk 4]
```diff
@@ -15,34 +16,20 @@ class OrderModel {
...
-        if(isUser.length > 0) {
-            userID = isUser[0].id;
-        }
-        else {
-            userID = await this.insertUser(user).then(result => result.rows[0].id);
-        };
-        
         // add order and return order id
-        orderID =  await this.insertOrder(userID).then(order => order.rows[0].id);
+        orderID = await this.insertOrder(user.id).then(order => order.rows[0].id);
```
Comment (ghost) [inline demo2/models/orderModel.js]: You can use `const` here.

**Review by ghost — COMMENTED** [review]
Please, add newline to the end of files.

File: demo2/common/midlwares/authMidlware.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+const orderModel = require('../../models/orderModel');
+const NotFound = require('../errors/notFound');
+
+module.exports = async (req,res,next) => {
+    const {phone, password } = req.headers;
+    const isUser = await orderModel.getUser(phone,password).then(user => user.rows);
```
Comment (kolesnicknick) [inline demo2/common/midlwares/authMidlware.js]: dont use await + then, choose 1 approach and stick to it

File: demo2/common/midlwares/authMidlware.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+    if (isUser.length > 0) {
+        res.locals.user = isUser[0];
+        res.locals.isAuthenticated = true;
+    }else {
+        res.locals.isAuthenticated = false;
+        next(new NotFound(401, "There is not such user in DB"));
```
Comment (kolesnicknick) [inline demo2/common/midlwares/authMidlware.js]: what happens if password is wrong?

File: demo2/common/midlwares/orderMidlware.js [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+module.exports = async (req,res,next) => {
+    const {products } = req.body;
+    const productsValid = await orderModel.productsValidation(products);
+    if (productsValid) {
+        res.locals.products = products;
+    }else {
+        next(new NotFound(404, "There is not such products in DB"));
+    }
+    next ();
+}
```
Comment (kolesnicknick) [inline demo2/common/midlwares/orderMidlware.js]: all this logic should be in the OrderModel

**Pull request merged**